cmd/thanos/tools.go

```diff
@@ -36,29 +38,41 @@ func registerCheckRules(app extkingpin.AppClause) {
 
 func checkRulesFiles(logger log.Logger, files *[]string) error {
 	var failed errutil.MultiError
+	errhandle := func(e error) {
+		level.Error(logger).Log("result", "FAILED", "error", e)
+		level.Info(logger).Log()
+		failed.Add(e)
+	}
 
 	for _, fn := range *files {
 		level.Info(logger).Log("msg", "checking", "filename", fn)
-		f, err := os.Open(fn)
+		matches, err := filepath.Glob(fn)
 		if err != nil {
-			level.Error(logger).Log("result", "FAILED", "error", err)
-			level.Info(logger).Log()
-			failed.Add(err)
+			errhandle(err)
+			continue
+		}
+		if matches == nil {
+			err = errors.New("matching file not found")
+			errhandle(err)
 			continue
 		}
-		defer func() { _ = f.Close() }()
+		for _, fn1 := range matches {
+			// no need to check path error
```

Comment:
Why not? :thinking:

Reply:
> Why not?

If the path is invalid then it wouldn't be in `matches` at the first place :thinking:

Reply:
Yes but we still may not be able to open that file for a lot of reasons (:

Reply:
> Yes but we still may not be able to open that file for a lot of reasons (:

makes sense :thinking: